### ISSUE TITLE
Improve Rendering of `<h1>` etc. in Docstrings

### ISSUE THREAD

**Issue by jacksund** [mitmproxy/pdoc#375]
This issue is just minor suggestions for the default HTML template used by pdoc. These are just preferential suggestions, so feel free to ignore or close this issue if you'd like!

1. Have the "built with pdoc" link open a new tab. When a link switches to a new website it's common practice to use `target="_blank"` to open a new tab rather than update the current one. This would be a super quick fix for pdoc:
``` html
<a href="..." target="_blank">built with pdoc</a>
``` 

2. Add left-side padding under each documentation header. Under a class method, the "property" or "returns" headers are often larger text than the method name itself -- so at a glance, it's difficult organize what belong to what. It'd be much nicer if left-side padding was added to the `<div>` of each class/method/property heading. For example:
``` html

<div>
<h3>MyExampleClass</h3>
    <div style=padding-left: 35px>
          <p>My rendered markdown docstring for this class!</p>
          <h3>def my_example_method(...)</h3>
               <div style=padding-left: 35px>
                     <p>My rendered markdown docstring for this method!</p>
               <div>
    <div>
</div>
```

I'd be happy to implement these as a pull-request if they're accepted!

**Comment by jacksund** [comment]
whoops, accidentally hit submit before I finished writing the issue. Sorry about that. I'm editting the initial comment above to fix this

**Comment by jacksund** [comment]
Updated issue has been added. Sorry about that.

**Comment by jacksund** [comment]
For suggestion 1:
Switch would only involve changing this one line:
https://github.com/mitmproxy/pdoc/blob/995f4f0c02ba71cdb459886346b72eede8d23464/pdoc/templates/default/module.html.jinja2#L64
It would change to:
``` html
<a class="attribution" title="pdoc: Python API documentation generator" href="https://pdoc.dev" target="_blank">
```

**Comment by mhils** [comment]
Absolutely no objections to 1), do you have a screenshot for 2)? I'm not exactly sure what you're referring to.

**Comment by jacksund** [comment]
# Current implementation

![image](https://user-images.githubusercontent.com/47992949/162525393-ed7633b4-1a0c-416d-8aef-e33961509ac1.png)

For the first time a user is ever looking at pdoc-style documentation, it's not immediately obvious where the "Parameters" section belongs to. In fact, it looks like "Parameters" is a new section header for the "def execute" entry. This is because the "Parameters" text size is larger than "def execute" and because both entry are at the same indent level. 

# Suggested implementation

![image](https://user-images.githubusercontent.com/47992949/162526512-0349c81c-06a1-4cbe-a489-413e75cbe1c0.png)

All I did here was add indents to each methods docstring (i.e. `<div style="padding-left:35px>`). At a glance, its much more obvious where the "Parameters" section belongs to.

This is also how sphinx makes their documentation easier to read. Here a random screenshot, where you can see the text is also indented under each method header: 

![image](https://user-images.githubusercontent.com/47992949/162526782-d4b612dd-6b2a-4f59-8bf0-2b01b059d76f.png)


Hopefully this helps!

**Comment by mhils** [comment]
Thanks! The example is helpful. I'll have to think about this a bit. I'm not totally opposed to adding the indent, but reducing the h1/h2/h3/h4/h5 font size in docstrings might be a reasonable alternative. We internally use h6 when converting from Numpy/Google-style docstrings, which shows that something's off here. It may be the right call to actually do both.

**Comment by jacksund** [comment]
Sweet! Both would be super helpful for method docstrings. In my own docs, I do the same thing by using `####` to get h6 size headings.

**Comment by mhils** [comment]
fixed in #383 ✨
